File: 1.two-sum.java
import java.util.HashMap;
import java.util.Map;

/*
 * @lc app=leetcode id=1 lang=java
 *
 * [1] Two Sum
 */

// @lc code=start
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> temp = new HashMap<>();
        int[] result = new int[2];
        for (int i = 0; i < nums.length; i++) {
            if (temp.containsKey(target - nums[i])) {
                result[0] = temp.get(target - nums[i]);
                result[1] = i;
            }
            temp.put(nums[i], i);
        }
        return result;
    }
}
// @lc code=end

File: Python/1.two-sum.py
#
# @lc app=leetcode id=1 lang=python3
#
# [1] Two Sum
#

# @lc code=start
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        x = {}
        for i, n in enumerate(nums):
            if target - n in x:
                return x[target - n], i
            x[n] = i
# @lc code=end

